presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5741,4 +5741,34 @@ public void testMapUnionSumOverflow()
                 "select y, map_union_sum(x) from (select 1 y, map(array['x', 'z', 'y'], cast(array[null,30, 32760] as array<smallint>)) x " +
                         "union all select 1 y, map(array['x', 'y'], cast(array[1,100] as array<smallint>))x) group by y", ".*Value 32860 exceeds MAX_SHORT.*");
     }
+
+    @Test
+    public void testLosslessJoin()
+    {
+        MaterializedResult countPart = computeActual("select partkey from part group by partkey");
+        MaterializedResult countLineitem = computeActual("select orderkey from lineitem group by orderkey");
+
+        MaterializedResult actual = computeActual("select p.partkey from part p left join lineitem l on p.partkey = l.orderkey group by p.partkey");
+        assertEquals(actual.getRowCount(), countPart.getRowCount());
+        assertNotEquals(actual.getRowCount(), countLineitem.getRowCount());
+
+        actual = computeActual("select l.orderkey from part p right join lineitem l on p.partkey = l.orderkey group by 1");
+        assertEquals(actual.getRowCount(), countLineitem.getRowCount());
+        assertNotEquals(actual.getRowCount(), countPart.getRowCount());
+
+        actual = computeActual("select p.partkey from part p left join lineitem l on p.partkey = l.orderkey");
+        assertNotEquals(actual.getRowCount(), countLineitem.getRowCount());
+        assertNotEquals(actual.getRowCount(), countPart.getRowCount());
+        assertEquals(actual.getRowCount(), 3500);
```

Comment:
This should be the same row count as `SELECT partkey from part`?

Reply:
no, this query is equivalent to `select distinct partkey from part` or with a `group by partkey` from table `part`